This really disables altivec if not wanted.
Patch from xtophe: https://trac.videolan.org/vlc/ticket/1055
https://bugs.gentoo.org/show_bug.cgi?id=158126
===================================================================
--- vlc-0.8.6_p18636.orig/configure.ac
+++ vlc-0.8.6_p18636/configure.ac
@@ -1304,7 +1304,7 @@ fi
 AC_CHECK_HEADERS(altivec.h)
 CPPFLAGS="${CPPFLAGS_save}"
 
-if test "${ac_cv_c_altivec}" != "no"; then
+if test "${ac_cv_c_altivec}" != "no" -a "${enable_altivec}" != "no"; then
   AC_DEFINE(CAN_COMPILE_C_ALTIVEC, 1, Define if your compiler groks C AltiVec extensions.)
   VLC_ADD_CFLAGS([vlc],[${ac_cv_c_altivec}])
   VLC_ADD_CFLAGS([idctaltivec motionaltivec],[${ac_cv_c_altivec}])
